src/Block/ChildrenPagesBlockService.php

```diff
@@ -26,33 +26,22 @@
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\OptionsResolver\OptionsResolver;
-use Symfony\Component\Templating\EngineInterface;
+use Twig\Environment;
 
 /**
  * Render children pages.
  *
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
- *
- * @final since sonata-project/page-bundle 3.26
  */
-class ChildrenPagesBlockService extends AbstractAdminBlockService
+final class ChildrenPagesBlockService extends AbstractBlockService
 {
-    /**
-     * @var SiteSelectorInterface
-     */
-    protected $siteSelector;
-
-    /**
-     * @var CmsManagerSelectorInterface
-     */
-    protected $cmsManagerSelector;
-
-    /**
-     * @param string $name
-     */
-    public function __construct($name, EngineInterface $templating, SiteSelectorInterface $siteSelector, CmsManagerSelectorInterface $cmsManagerSelector)
```

Comment:
This can't be changed in a BC way without a lot of trouble. Probably it can be done, but it is not worth it, the idea is to release 4.0 as soon as possible.

Reply:
I will wait for @VincentLanglet confirmation before do the complete review, But in the end what I think that could be a problem is just add `type` in the code and change the property visibility like from `protected` to `private`.

I'm just thinking in case someone extends of those class.

Reply:
- and for remove property without `@deprecation` annotation?
- and for add or remove params in `__construct` or other `method`, is it fine too?

Reply:
Ok, thanks for the explanation :)

Reply:
So, for that review? there is no problem?

Reply:
Except my review on SnapshotChildrenCollection

Reply:
for me it's good :)

Reply:
So, what can I do?